src/sim_engine/engine.rs

```diff
@@ -71,12 +72,19 @@ impl Engine for SimEngine {
     }
 
     fn destroy_pool(&mut self, name: &str) -> EngineResult<()> {
+        let entry = match self.pools.entry(name.into()) {
+            Entry::Vacant(_) => return Ok(()),
```

Comment:
I tried both. Considerations of lifetimes seemed to make the ```get()``` approach just as hard to read. So I went with the entry approach because it was more efficient.